cmd/gateway/zerostor/tlog/filesystem_meta.go

```diff
@@ -0,0 +1,433 @@
+	if err != nil {
+		return info, err
+	}
+
+	_, err = t.recorder.Record(Record{
+		OperationUploadComplete,
+		bucket,
+		object,
+		uploadID,
+		partsBytes,
+	}, true)
+	return info, err
+}
+
+// GetObjectInfo returns info about a bucket object
+func (t *fsTLogger) GetObjectInfo(bucket, object string) (minio.ObjectInfo, error) {
+	return t.meta.GetObjectInfo(bucket, object)
+}
+
+// StreamObjectMeta streams an object metadata blobs through a channel
+func (t *fsTLogger) StreamObjectMeta(ctx context.Context, bucket, object string) <-chan meta.Stream {
+	return t.meta.StreamObjectMeta(ctx, bucket, object)
+}
+
+//WriteMetaStream writes all the incomming meta stream
+func (t *fsTLogger) WriteMetaStream(ctx context.Context, c <-chan *metatypes.Metadata, bucket, object string) <-chan error {
+	errc := make(chan error)
+	go func() {
+		var totalSize int64
+		var modTime int64
+		var previousPart meta.ObjectMeta
+		var firstPart meta.ObjectMeta
+		var objMeta meta.ObjectMeta
+		counter := 0
+
+		defer close(errc)
+		for metaData := range c {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+
+			totalSize += metaData.Size
+			modTime = metaData.LastWriteEpoch
+			objMeta := meta.ObjectMeta{
+				Metadata: *metaData,
+				Filename: uuid.NewV4().String(),
+			}
+
+			if counter > 0 {
+				previousPart.NextBlob = objMeta.Filename
+				if err := t.WriteObjMeta(&previousPart); err != nil {
+					errc <- err
+				}
+				if counter == 1 {
+					// link the first blob
+					firstPart = previousPart
+					if err := t.LinkObject(bucket, object, firstPart.Filename); err != nil {
+						errc <- err
+					}
+				}
+			}
+			previousPart = objMeta
+			counter++
+		}
+
+		// write the meta of the last received metadata
+		if err := t.WriteObjMeta(&objMeta); err != nil {
+			errc <- err
+		}
+		firstPart.ObjectSize = totalSize
+		firstPart.ObjectModTime = modTime
+		firstPart.ObjectUserMeta = firstPart.UserDefined
+
+		// update the the first meta part with the size and mod time
+		if err := t.WriteObjMeta(&firstPart); err != nil {
+			errc <- err
+		}
+	}()
+	return errc
+}
+
+// StreamBlobs streams all blobs
+func (t *fsTLogger) StreamBlobs(ctx context.Context) <-chan meta.Stream {
+	return t.meta.StreamBlobs(ctx)
+}
+
+// ValidUpload checks if an upload id is valid
+func (t *fsTLogger) ValidUpload(bucket, uploadID string) (bool, error) {
+	return t.meta.ValidUpload(bucket, uploadID)
+}
+
+// ListMultipartUploads lists multipart uploads that are in progress
+func (t *fsTLogger) ListMultipartUploads(bucket string) (minio.ListMultipartsInfo, error) {
+	return t.meta.ListMultipartUploads(bucket)
+}
+
+// StreamMultiPartsMeta streams parts metadata for a multiupload
+func (t *fsTLogger) StreamMultiPartsMeta(ctx context.Context, bucket, uploadID string) <-chan meta.Stream {
+	return t.meta.StreamMultiPartsMeta(ctx, bucket, uploadID)
+}
+
+//Sync syncs the backend storage with the latest records from the tlog storage
```

Comment:
The code in HealthCheck and Sync can be extracted in functions to be used if we ever needed to add other tlog meta managers. I just didnt want to have another struct wrap the tlog struct and pass the recorder to both structs and end up with lots of redundancy like before.